### PR TITLE
ceph: Update pg_num related documents

### LINKED ISSUE #1329: docs: explain "too few PGs per OSD" warning
In a lot of scenarios, the ceph status will show something like `too few PGs per OSD (25 < min 30)`, which can be fairly benign.  The consequences of too few PGs is much less severe than the consequences of too **many** PGs, as explained in this ceph blog post:
http://ceph.com/community/new-luminous-pg-overdose-protection/
> If you have way less than that the load distribution won’t be as balanced (some OSDs will store more than others), and if you have too many more than that then each OSD will be coordinating with too many other OSDs, increasing overhead and reducing overall reliability.

We should mention this in the docs/FAQs somewhere.

### PR DESCRIPTION
**Pull request by satoru-takeuchi**
**Description of your changes:**

Update pg_num related documents.

**Which issue is resolved by this Pull Request:**

Resolves #1329

**Checklist:**

- [o] **CommitLint Bot**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [o] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [o] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [o] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [o] Documentation has been updated, if necessary.
- [o] Unit tests have been added, if necessary.
- [o] Integration tests have been added, if necessary.
- [o] Pending release notes updated with breaking and/or notable changes, if necessary.
- [o] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [o] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[skip ci]

### CODE REVIEW

File: Documentation/ceph-common-issues.md [hunk 1]
```diff
@@ -768,3 +768,15 @@ This problem will be solve by the following two fixes.
 * Ceph: A fix that uses the above-mentioned kernel's feature. The Ceph community will probably discuss this fix after releasing Linux v5.6.
 
 You can bypass this problem by using ext4 or any other filesystems rather than XFS. Filesystem type can be specified with `csi.storage.k8s.io/fstype` in StorageClass resource.
+
+## Too few PGs per OSD warning is shown
```
Comment (travisn) [inline Documentation/ceph-common-issues.md:772]: Please add this to the table of contents links at the top of this topic so it is more discoverable

File: Documentation/ceph-common-issues.md [hunk 2]
```diff
@@ -768,3 +768,15 @@ This problem will be solve by the following two fixes.
...
+
+### Symptoms
+
+- `ceph status` shows "too few PGs per OSD" warning.
```
Comment (travisn) [inline Documentation/ceph-common-issues.md:776]: How about showing a snippet of the output from `ceph status` in a multi-line comment so it's clear that this is a ceph health warning?

File: Documentation/ceph-common-issues.md [hunk 3]
```diff
@@ -768,3 +768,15 @@ This problem will be solve by the following two fixes.
...
+
+### Solution
+
+The meaning of this warning is written in [the document](https://docs.ceph.com/docs/master/rados/operations/health-checks#too-few-pgs).
+However, in many case, it's venign. For more information, please see [the blog entry](http://ceph.com/community/new-luminous-pg-overdose-protection/).
```
Comment (travisn) [inline Documentation/ceph-common-issues.md:781]:
```suggestion
However, in many case, it's benign. For more information, please see [the blog entry](http://ceph.com/community/new-luminous-pg-overdose-protection/).
```

File: Documentation/ceph-configuration.md [hunk 4]
```diff
@@ -45,11 +49,16 @@ spec:
...
+In Octopus (v15.2.x), this module is enabled by default without the above-mentioned setting.
+
 With that setting, the autoscaler will be enabled for all new pools. If you do not desire to have
-the autoscaler enabled for all pools, you will need to use the Rook toolbox to enable the module
+the autoscaler enabled for all new pools, you will need to use the Rook toolbox to enable the module
 and [enable the autoscaling](https://docs.ceph.com/docs/master/rados/operations/placement-groups/)
 on individual pools.
 
+The autoscaler is not enabled for the existing pools after enabling the module. So if you want to
+enable the autoscaling for these pools, you should do so by yourself.
```
Comment (travisn) [inline Documentation/ceph-configuration.md:60]:
```suggestion
enable the autoscaling for these existing pools, they must be configured from the toolbox.
```

File: Documentation/ceph-common-issues.md [hunk 5]
```diff
@@ -768,3 +768,15 @@ This problem will be solve by the following two fixes.
...
+Please refer to [Configuring Pools](ceph-advanced-configuration.md#configuring-pools) if you want to know the proper pg_num of pools and change these values.
```
Comment (leseb) [inline Documentation/ceph-common-issues.md:782]:
```suggestion
Please refer to [Configuring Pools](ceph-advanced-configuration.md#configuring-pools) if you want to know the proper `pg_num` of pools and change these values. 
```

File: Documentation/ceph-common-issues.md (hunk 1 above)
Comment (satoru-takeuchi) [reply Documentation/ceph-common-issues.md:773]: Added, thanks.

File: Documentation/ceph-common-issues.md (hunk 2 above)
Comment (satoru-takeuchi) [reply Documentation/ceph-common-issues.md]: Added the result of `ceph status` as you said.

File: Documentation/ceph-common-issues.md (hunk 3 above)
Comment (satoru-takeuchi) [reply Documentation/ceph-common-issues.md]: Ah, thank for finding my typo. Fixed.

File: Documentation/ceph-configuration.md (hunk 4 above)
Comment (satoru-takeuchi) [reply Documentation/ceph-configuration.md]: Your suggestion makes sense. Fixed.

File: Documentation/ceph-common-issues.md (hunk 5 above)
Comment (satoru-takeuchi) [reply Documentation/ceph-common-issues.md]: Fixed, thanks.

File: Documentation/ceph-common-issues.md [hunk 6]
```diff
@@ -768,3 +769,23 @@ This problem will be solve by the following two fixes.
...
+  cluster:
+    id:     fd06d7c3-5c5c-45ca-bdea-1cf26b783065
+    health: HEALTH_WARN
+            too few PGs per OSD (16 < min 30)
+```
+
+### Solution
+
+The meaning of this warning is written in [the document](https://docs.ceph.com/docs/master/rados/operations/health-checks#too-few-pgs).
+However, in many case, it's benign. For more information, please see [the blog entry](http://ceph.com/community/new-luminous-pg-overdose-protection/).
```
Comment (travisn) [inline Documentation/ceph-common-issues.md:790]:
```suggestion
However, in many cases it is benign. For more information, please see [the blog entry](http://ceph.com/community/new-luminous-pg-overdose-protection/).
```

Comment (satoru-takeuchi) [reply Documentation/ceph-common-issues.md]: Fixed, thanks.

**Pull request merged**